disco_aws_automation/disco_deploy.py

```diff
@@ -436,8 +436,30 @@ def handle_blue_green_ami(self, pipeline_dict, ami, old_group,
         new_group_config["min_size"] = min_size
         new_group_config["max_size"] = max_size
 
-        # Spinup our new autoscaling group in testing mode, making one even if one already exists.
-        self._disco_aws.spinup([new_group_config], create_if_exists=True, testing=True)
+        try:
+            # Spinup our new autoscaling group in testing mode, making one even if one already exists.
+            self._disco_aws.spinup([new_group_config], create_if_exists=True, testing=True)
+        except Exception:
+            logging.exception("Spinning up a new autoscaling group failed")
+
+            # Try to grab the new group. If it exists, we get a group. If not, we get a `None`.
+            new_group = self._disco_autoscale.get_existing_group(hostclass=hostclass,
```

Comment:
That shouldn't happen. It should return the most recently created ASG, and we check to make sure that it isn't the original ASG. So if we get an ASG out of this, it should be the most recently created ASG. Which we would want to delete.